rllib/env/single_agent_env_runner.py

```diff
@@ -498,25 +479,12 @@ def _sample_episodes(
             ts += self.num_envs
 
             for env_index in range(self.num_envs):
-                extra_model_output = {k: v[env_index] for k, v in to_env.items()}
-
-                if terminateds[env_index] or truncateds[env_index]:
+                if _was_terminated[env_index] or _was_truncated[env_index]:
                     eps += 1
 
-                    episodes[env_index].add_env_step(
-                        infos[env_index].pop("final_observation"),
```

Comment:
Okay, i.e. with `gymnasium>=1.0.0` the `final_observation` is gone and instead a regular observartion will be returned?

Reply:
Correct, the final observation is returned in the actual `obs`. The reset obs, you only get on the next(!) call to step, together with a dummy reward of 0.0.